driver-legacy/src/main/com/mongodb/MongoClientOptions.java

```diff
@@ -1009,6 +1029,7 @@ public static class Builder {
         private final List<ConnectionPoolListener> connectionPoolListeners = new ArrayList<ConnectionPoolListener>();
         private final List<ServerListener> serverListeners = new ArrayList<ServerListener>();
         private final List<ServerMonitorListener> serverMonitorListeners = new ArrayList<ServerMonitorListener>();
+        private final ConnectionPoolSettings connectionPoolSettings = ConnectionPoolSettings.builder().build();
```

Comment:
It's confusing to make this a field.   Just move the initialization of the three connection pool-related fields to the constructor so that this can just be a local variable.